File: 330. Patching Array/Python/Solution.py
class Solution(object):
    def minPatches(self, nums, n):
        """
        :type nums: List[int]
        :type n: int
        :rtype: int
        """
        hope = 1
        result = 0
        index = 0
        while hope <= n:
            if index < len(nums) and nums[index] <= hope:
                hope += nums[index]
                index += 1
            else:
                result += 1
                hope += hope
        return result

File: 330. Patching Array/C++/Solution.cpp
class Solution {
public:
    int minPatches(vector<int> &nums, int n) {
        int hope = 1;
        int result = 0;
        int index = 0;
        while (hope <= n) {
            if (index < nums.size() && nums[index] <= hope) {
                hope += nums[index];
                index++;
            } else {
                result++;
                if (n - hope < hope) {
                    break;
                }
                hope += hope;
            }
        }
        return result;
    }
};